src/litdata/streaming/client.py

```diff
@@ -1,70 +1,70 @@
 
 
-class S3Client:
-    # TODO: Generalize to support more cloud providers.
+# class S3Client:
+#     # TODO: Generalize to support more cloud providers.
 
-    def __init__(self, refetch_interval: int = 3300, storage_options: Optional[Dict] = {}) -> None:
-        self._refetch_interval = refetch_interval
-        self._last_time: Optional[float] = None
-        self._client: Optional[Any] = None
-        self._storage_options: dict = storage_options or {}
+#     def __init__(self, refetch_interval: int = 3300, storage_options: Optional[Dict] = {}) -> None:
+#         self._refetch_interval = refetch_interval
+#         self._last_time: Optional[float] = None
+#         self._client: Optional[Any] = None
+#         self._storage_options: dict = storage_options or {}
 
-    def _create_client(self) -> None:
-        has_shared_credentials_file = (
-            os.getenv("AWS_SHARED_CREDENTIALS_FILE") == os.getenv("AWS_CONFIG_FILE") == "/.credentials/.aws_credentials"
-        )
+#     def _create_client(self) -> None:
+#         has_shared_credentials_file = (
+#           os.getenv("AWS_SHARED_CREDENTIALS_FILE") == os.getenv("AWS_CONFIG_FILE") == "/.credentials/.aws_credentials"
+#         )
 
-        if has_shared_credentials_file or not _IS_IN_STUDIO:
-            self._client = boto3.client(
-                "s3",
-                **{
-                    "config": botocore.config.Config(retries={"max_attempts": 1000, "mode": "adaptive"}),
-                    **self._storage_options,
-                },
-            )
-        else:
-            provider = InstanceMetadataProvider(iam_role_fetcher=InstanceMetadataFetcher(timeout=3600, num_attempts=5))
-            credentials = provider.load()
-            self._client = boto3.client(
-                "s3",
-                aws_access_key_id=credentials.access_key,
-                aws_secret_access_key=credentials.secret_key,
-                aws_session_token=credentials.token,
-                config=botocore.config.Config(retries={"max_attempts": 1000, "mode": "adaptive"}),
-            )
+#         if has_shared_credentials_file or not _IS_IN_STUDIO:
+#             self._client = boto3.client(
+#                 "s3",
+#                 **{
+#                     "config": botocore.config.Config(retries={"max_attempts": 1000, "mode": "adaptive"}),
+#                     **self._storage_options,
+#                 },
+#             )
+#         else:
+#           provider = InstanceMetadataProvider(iam_role_fetcher=InstanceMetadataFetcher(timeout=3600, num_attempts=5))
+#             credentials = provider.load()
+#             self._client = boto3.client(
+#                 "s3",
+#                 aws_access_key_id=credentials.access_key,
+#                 aws_secret_access_key=credentials.secret_key,
+#                 aws_session_token=credentials.token,
+#                 config=botocore.config.Config(retries={"max_attempts": 1000, "mode": "adaptive"}),
```

Comment:
Can we make this we have those defaults for S3 client and it is additive to whather the user adds as extra. 

```
default_storage_kwargs = {retries=retries={"max_attempts": 1000, "mode": "adaptive"}}
storage_options = {**default_storage_kwargs, storage_options}
```

Reply:
```python
_DEFAULT_STORAGE_OPTIONS = {
    "s3":{"config_kwargs": {"retries":{"max_attempts": 1000, "mode": "adaptive"}}},
}

def get_complete_storage_options(cloud_provider: str, storage_options: Optional[Dict] = {}) -> Dict:
    if storage_options is None:
        storage_options = {}
    if cloud_provider in _DEFAULT_STORAGE_OPTIONS:
        return {**_DEFAULT_STORAGE_OPTIONS[cloud_provider], **storage_options}
    return storage_options
```

- Only adding for S3 now. `gcsfs` doesn't seem to have such retry storage_option. But, if it has we can extend from here on in future.